Add notes for DEX (#181)
epinio/docs: docs/references/authentication_oidc.md

Since version **1.3.0**, Epinio has integrated [Dex](https://dexidp.io/) as an identity provider which adds the support for external OIDC providers.

:::caution

> In order to make Dex work the ingress address (set by helm `global.domain` value) must be reachable also from kubernetes pods. Always use either FQDN hostname or a public IP address together with some wildcard DNS service as `nip.io` for epinio ingresses. Using localhost or entries in /etc/hosts is not sufficient.

Reviewer: ```suggestion
In order to make Dex work the ingress address (set by the chart's `global.domain` value) has to be reachable from the kubernetes pods, not just from outside. Always use either a FQDN hostname or a public IP address together with some wildcard DNS service as `xip.io` for epinio ingresses. Using localhost or entries in `/etc/hosts` is __not__ sufficient.
```

Note: IIRC `nip...` was a slur in some contexts.

Author: Thanks, suggestion applied.

:::

To authenticate through Dex, you can use the login command with the `--oidc` flag. This will open a web page where you can authenticate with the configured providers.

```bash
epinio login --oidc https://epinio.mydomain.com
```

> If you are using the `epinio` cli on a machine without a browser you can provide the `--prompt` flag. This will give you the url of a web page where you can authenticate even on different machine. After logging in and pressing `Grant Access` button the page will return the authorization code that you should copy and paste back to the `epinio` cli input to finish the authentication process.

Reviewer: ```suggestion
If you are using the `epinio` cli on a machine without a browser you can provide the `--prompt` flag. This will give you the url of a web page where you can authenticate even on a different machine. After logging in and pressing the `Grant Access` button the page will return the authorization code that you have to copy and paste back to the `epinio` cli input to finish the authentication process.
```

Author: Thanks, applied.
